diff --git a/decode.js b/decode.js
index 8d32bd36435c7daf57ca41025e08783d70fd420b..0ca8df415119c0a1443361a0271db7fccabb8cbd 100644
--- a/decode.js
+++ b/decode.js
@@ -14,6 +14,11 @@ import { initEmscriptenModule } from './utils.js';
 import mozjpeg_dec from './codec/dec/mozjpeg_dec.js';
 let emscriptenModule;
 export async function init(module, moduleOptionOverrides) {
+    if (typeof module === 'string') {
+        module = await fetch(module)
+            .then((response) => response.arrayBuffer())
+            .then((bytes) => new WebAssembly.Module(bytes))
+    }
     emscriptenModule = initEmscriptenModule(mozjpeg_dec, module, moduleOptionOverrides);
 }
 export default async function decode(buffer) {
