diff --git a/encode.js b/encode.js
index e7df0fd448300fe398e82b2e8d9c73e6702ec87d..ecec668b9bb12d2a1026317cc2e56e676e82f3c3 100644
--- a/encode.js
+++ b/encode.js
@@ -15,6 +15,11 @@ import { initEmscriptenModule } from './utils';
 import { simd } from 'wasm-feature-detect';
 let emscriptenModule;
 export async function init(module) {
+    if (typeof module === 'string') {
+        module = await fetch(module)
+            .then((response) => response.arrayBuffer())
+            .then((bytes) => new WebAssembly.Module(bytes))
+    }
     if (await simd()) {
         const webpEncoder = await import('./codec/enc/webp_enc_simd');
         emscriptenModule = initEmscriptenModule(webpEncoder.default, module);
