diff --git a/encode.js b/encode.js
index c23ed8b6871659c8c3ed1b834f6d87823bad0588..532c5992ec567928263bbf8e64454abc6e768166 100644
--- a/encode.js
+++ b/encode.js
@@ -15,6 +15,11 @@ import { initEmscriptenModule } from './utils.js';
 import { simd } from 'wasm-feature-detect';
 let emscriptenModule;
 export async function init(module, moduleOptionOverrides) {
+    if (typeof module === 'string') {
+        module = await fetch(module)
+            .then((response) => response.arrayBuffer())
+            .then((bytes) => new WebAssembly.Module(bytes))
+    }
     if (await simd()) {
         const webpEncoder = await import('./codec/enc/webp_enc_simd.js');
         emscriptenModule = initEmscriptenModule(webpEncoder.default, module, moduleOptionOverrides);
