diff --git a/dist/compiler/server/plugins/bareImports.js b/dist/compiler/server/plugins/bareImports.js
index bb0dfdb4de0304fe105a4111be05a02c2baef5f4..57fd316a730840ccb01a10a110885ab9b349a808 100644
--- a/dist/compiler/server/plugins/bareImports.js
+++ b/dist/compiler/server/plugins/bareImports.js
@@ -74,9 +74,6 @@ function serverBareModulesPlugin(ctx) {
             throw e;
           }
         }
-        if (loader === "file") {
-          return undefined;
-        }
 
         // Always bundle CSS side-effect imports.
         if (cssSideEffectImports.isCssSideEffectImportPath(path)) {
@@ -89,9 +86,18 @@ function serverBareModulesPlugin(ctx) {
           // bundle it if the path matches the pattern
           if (typeof pattern === "string" ? path === pattern : pattern.test(path)) {
             return undefined;
+          } else {
+            return {
+              path,
+              external: true
+            };
           }
         }
 
+        if (loader === "file") {
+          return undefined;
+        }
+
         // Externalize everything else if we've gotten here.
         return {
           path,
